libnmstate/nm/sriov.py

```diff
@@ -162,42 +140,69 @@ def get_info(device):
     except FileNotFoundError:
         return info
 
-    connection = nm_connection.ConnectionProfile()
-    connection.import_by_device(device)
-    if not connection.profile:
-        info[Ethernet.SRIOV_SUBTREE] = sriov_config
-        return info
-
-    sriov_setting = connection.profile.get_setting_by_name(
-        nmclient.NM.SETTING_SRIOV_SETTING_NAME
-    )
-
-    if sriov_setting:
-        vfs_config = _get_info_sriov_vfs_config(sriov_setting)
-        sriov_config[Ethernet.SRIOV.VFS_SUBTREE] = vfs_config
+    if sriov_config[Ethernet.SRIOV.TOTAL_VFS]:
+        sriov_config[
+            Ethernet.SRIOV.VFS_SUBTREE
+        ] = _get_sriov_vfs_configuration(ifname)
+    else:
+        sriov_config[Ethernet.SRIOV.VFS_SUBTREE] = []
 
-        info[Ethernet.SRIOV_SUBTREE] = sriov_config
+    info[Ethernet.SRIOV_SUBTREE] = sriov_config
 
     return info
 
 
-def _get_info_sriov_vfs_config(sriov_setting):
-    vfs_config = []
-    vfs_setting = sriov_setting.props.vfs
-    for vf in vfs_setting:
-        vf_config = {}
-        vf_config[Ethernet.SRIOV.VFS.ID] = vf.get_index()
-        for nm_attribute in SRIOV_NM_TO_NMSTATE_MAP.keys():
-            _get_nm_attribute(vf, vf_config, nm_attribute)
+def _get_sriov_vfs_configuration(ifname):
```

Comment:
IMHO, in nmstate, the `configuration` means on-disk profile/connection. So please use `_get_sriov_vfs_info`.

Reply:
Done, thanks!